agent/artifact_batch_creator.go

```diff
@@ -86,7 +86,8 @@ func (a *ArtifactBatchCreator) Create(ctx context.Context) ([]*api.Artifact, err
 			}
 
 			creation, resp, err := a.apiClient.CreateArtifacts(ctxTimeout, a.conf.JobID, batch)
-			if resp != nil && (resp.StatusCode == 401 || resp.StatusCode == 404) {
+			// the server returns a 403 code if the artifact has exceeded the service quota
+			if resp != nil && (resp.StatusCode == 401 || resp.StatusCode == 403 || resp.StatusCode == 404) {
```

Comment:
It might be worth doing some debug printing to see what the value of `resp` is.

Something like
```suggestion
      a.logger.Info("response: %#v", resp)
			if resp != nil && (resp.StatusCode == 401 || resp.StatusCode == 403 || resp.StatusCode == 404) {
```

will give you a lot of information.